plugin/hook/adapter_test.go

```diff
@@ -0,0 +1,622 @@
+package hook
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	pb "github.com/odpf/optimus/api/proto/odpf/optimus"
+	"github.com/odpf/optimus/models"
+)
+
+func Test_adaptQuestionToProto(t *testing.T) {
+	t.Run("should work for empty inputs", func(t *testing.T) {
+		actual := adaptQuestionToProto(models.PluginQuestion{})
+		expected := &pb.PluginQuestion{}
+		assert.Equal(t, expected, actual)
+	})
+	t.Run("should work for non empty inputs (with empty SubQuestion parameter)", func(t *testing.T) {
+		actual := adaptQuestionToProto(models.PluginQuestion{
+			Name:         "key",
+			Prompt:       "sec",
+			Help:         "do",
+			Default:      "this",
+			Multiselect:  nil,
+			SubQuestions: []models.PluginSubQuestion{},
+		})
+		expected := &pb.PluginQuestion{
+			Name:         "key",
+			Prompt:       "sec",
+			Help:         "do",
+			Default:      "this",
+			Multiselect:  nil,
+			SubQuestions: nil,
+		}
+		assert.Equal(t, expected, actual)
+	})
+	t.Run("should work for nil inputs (with empty questions parameter)", func(t *testing.T) {
+		actual := adaptQuestionToProto(models.PluginQuestion{
+			Name:        "nil",
+			Prompt:      "nil",
+			Help:        "nil",
+			Default:     "nil",
+			Multiselect: nil,
+			SubQuestions: []models.PluginSubQuestion{
+				{
+					IfValue:   "",
+					Questions: []models.PluginQuestion{},
+				},
+			},
+		})
+		expected := &pb.PluginQuestion{
+			Name:        "nil",
+			Prompt:      "nil",
+			Help:        "nil",
+			Default:     "nil",
+			Multiselect: nil,
+			SubQuestions: []*pb.PluginQuestion_SubQuestion{
+				{
+					IfValue:   "",
+					Questions: []*pb.PluginQuestion{},
+				},
+			},
+		}
+		assert.Equal(t, expected, actual)
+	})
+	t.Run("should work for non empty inputs", func(t *testing.T) {
+		actual := adaptQuestionToProto(models.PluginQuestion{
+			Name:        "key",
+			Prompt:      "sec",
+			Help:        "this",
+			Default:     "do",
+			Multiselect: []string{"a", "ba", "b"},
+			SubQuestions: []models.PluginSubQuestion{
+				{
+					IfValue: "",
+					Questions: []models.PluginQuestion{
+						{
+							Name:         "nil",
+							Prompt:       "nil",
+							Help:         "nil",
+							Default:      "nil",
+							Multiselect:  nil,
+							SubQuestions: nil,
+						},
+					},
+				},
+			},
+		})
+		expected := &pb.PluginQuestion{
+			Name:        "key",
+			Prompt:      "sec",
+			Help:        "this",
+			Default:     "do",
+			Multiselect: []string{"a", "ba", "b"},
+			SubQuestions: []*pb.PluginQuestion_SubQuestion{
+				{
+					IfValue: "",
+					Questions: []*pb.PluginQuestion{
+						{
+							Name:         "nil",
+							Prompt:       "nil",
+							Help:         "nil",
+							Default:      "nil",
+							Multiselect:  nil,
+							SubQuestions: nil,
+						},
+					},
+				},
+			},
+		}
+		assert.Equal(t, expected, actual)
+	})
+	t.Run("should work for non empty inputs", func(t *testing.T) {
```

Comment:
whats the difference between this and line 66 test

Reply:
On line 66 I have passes nil to the Question object while in 133 it is any random para (non-zero)